lib/ServiceManager.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * 
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+
+import {
+  TokenManagerOptions,
+  ServiceManagerInterface
+} from './types';
+import { TokenManager } from './TokenManager';
+import {
+  BroadcastChannel,
+  createLeaderElection,
+  LeaderElector
+} from 'broadcast-channel';
+import { TokenService, AutoRenewService, SyncStorageService } from './services';
+import { isBrowser } from './features';
+
+export class ServiceManager implements ServiceManagerInterface {
+  protected tokenManager: TokenManager;
+  protected options: TokenManagerOptions;
+  private services: Map<string, TokenService>;
+  private channel?: BroadcastChannel;
+  private elector?: LeaderElector;
+  private started: boolean;
+
+  private static knownServices = ['autoRenew', 'syncStorage'];
+
+  constructor(tokenManager: TokenManager, options: TokenManagerOptions = {}) {
```

Comment:
ServiceManager could accept an instance of `OktaAuth` (like the AuthStateManager does https://github.com/okta/okta-auth-js/blob/master/lib/AuthStateManager.ts#L53). This would still give it access to the `tokenManager` and it's config options, but also future-proofs the ServiceManager, in case we need to add a Service in the future that needs different configs

```javascript
constructor(sdk: OktaAuth) {
  this.sdk = sdk;
  this.tokenManager = sdk.tokenManager;
  ...
}
```